### PR TITLE
Simplify rclnodejs/index.js

### LINKED ISSUE #736: rclnodejs/index.js could be simplified
The file `rclnodejs/index.js` could be simplified using `async function`s, making it easier to maintain. Other ways to simplify it would be array handling (with `Array.some()` and `Array.every()`) and replacing `let` with `const` here and there.

### PR DESCRIPTION
**Pull request by felixdivo**
Fixes #736.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -344,22 +350,15 @@ let rcl = {
...
-          debug('Finish regeneration.');
-          resolve();
-        })
-        .catch((e) => {
-          reject(e);
-        });
-    });
+    debug('Begin regeneration of JavaScript code from ROS IDL files...');
+
+    await generator.generateAll(true);
```
Comment (felixdivo) [inline index.js:359]: Does this need to happen in sequence or could this also be done "concurrently" with `Promise.all([...])`?

Comment (minggangw) [reply index.js:359]: The ts generator depends on the JS files generated by `generator.generateAll(true)`, @wayneparrott please help to take a look, thanks!

Comment (wayneparrott) [reply index.js:359]: @minggangw @felixdivo Yes, message generation is a 2 phase process: 1) generate js message files, 2) generate the interfaces.d.ts declaration file from the js message files.

Comment (felixdivo) [reply index.js:359]: Okay, thanks. Then this PR should be done. I'll just add a comment about the above.

Comment (felixdivo) [reply index.js:359]:
```suggestion
    # generate the messages and type declarations, which must be done in sequence
    await generator.generateAll(true);
```

**Comment by felixdivo** [comment]
I don't get why AppVeyor is failing though. But maybe it will go away now. Let's see.

File: index.js [hunk 2]
```diff
@@ -344,22 +350,16 @@ let rcl = {
...
-          debug('Finish regeneration.');
-          resolve();
-        })
-        .catch((e) => {
-          reject(e);
-        });
-    });
+    debug('Begin regeneration of JavaScript code from ROS IDL files...');
+
+    # generate the messages and type declarations, which must be done in sequence
```
Comment (minggangw) [inline index.js]: I think this line causes the failure on ci :)

Comment (felixdivo) [reply index.js]: Oh WTF, too much Python ... :sweat_smile:

**Comment by felixdivo** [comment]
> I don't get why AppVeyor is failing though. But maybe it will go away now. Let's see.

It didn't. [This](https://ci.appveyor.com/project/minggangw/rclnodejs/builds/36461095#L475) war the failing test I meant (@minggangw ;)). Seems like `node ./scripts/compile_tests.js` fails. I can't see why this would be cause by my changes. It appears that a few lines above, this shows up:

```
> node scripts/generate_messages.js
Start JavaScript message generation...
Incomplete service: test_interface_files.srv.Arrays.
Incomplete service: test_interface_files.srv.BasicTypes.
Incomplete service: test_interface_files.srv.Empty.
Generation complete.
```

But that also happens on the main branch on [AppVeyor](https://ci.appveyor.com/project/minggangw/rclnodejs/builds/36440411#L436) and probably also Travis CI.

**Comment by minggangw** [comment]
I've re-triggered the appveyor,  my experience is that the appveyor is not stable sometimes and you have to retry it. Hope it will pass this time and I will merge this PR once it's done, thanks!

**Review by minggangw — APPROVED** [review]
Thanks for making the code more readable, lgtm

**Pull request merged**